File: Gestion A C/models/Fournisseur.php
<?php
require_once 'Personne.php';
require_once 'Approvisionnement.php';
require_once 'Sys.php';

class Fournisseur extends Personne {
    private static int $nbF = 0;
    private string $fix;
    private array $approvisionnements = [];

    public function __construct() {
        parent::__construct();
        self::$nbF++;
        $this->id = "F-" . self::$nbF;
    }

    public function getFix(): string {
        return $this->fix;
    }

    public function setFix(string $fix): void {
        $this->fix = $fix;
    }

    public function getApprovisionnements(): array {
        return $this->approvisionnements;
    }

    public function addApprovisionnement(Approvisionnement $approvisionnement): void {
        $this->approvisionnements[] = $approvisionnement;
    }

    public function __toString(): string {
        return "\n---------------------------------------\n" .
            "ID        : " . $this->id . "\n" .
            "Nom       : " . Sys::capitalize($this->nom) . "\n" .
            "Prenom    : " . Sys::capitalize($this->prenom) . "\n" .
            "Telephone : " . $this->telephone . "\n" .
            "Fix       : " . $this->fix . "\n" .
            "Adresse   : " . Sys::capitalize($this->adresse);
    }
}

File: Gestion Atelier De Couture/src/entity/Fournisseur.java
package entity;

import java.util.ArrayList;

import services.Sys;

public class Fournisseur extends Personne {

    private String fix;
    private static int nbF;
    private ArrayList<Approvisionnement> approvisionnements = new ArrayList<>();

    public Fournisseur() {
        super();
        nbF++;
        this.id = "F-" + nbF;
    }

    public void setFix(String fix) {
        this.fix = fix;
    }

    public String getFix() {
        return fix;
    }

    public ArrayList<Approvisionnement> getApprovisionnements() {
        return approvisionnements;
    }

    public void addApprovisionnement(Approvisionnement approvisionnement) {
        this.approvisionnements.add(approvisionnement);
    }
 
    public String toString(){
        return "\n---------------------------------------\n" +
                "ID        : " + id + "\n" +
                "Nom       : " + Sys.capitalize(nom) + "\n" +
                "Prenom    : " + Sys.capitalize(prenom) + "\n" +
                "Telephone : " + telephone + "\n" +
                "Fix       : " + fix + "\n" +
                "Adresse   : " + Sys.capitalize(adresse);
    }
}
